spec/system/hashtag_autocomplete_spec.rb

```diff
@@ -4,31 +4,52 @@
          type: :system,
          js: true do
   fab!(:user) { Fabricate(:user) }
-  fab!(:topic) { Fabricate(:topic) }
+  fab!(:category) do
+    Fabricate(:category, name: "Cool Category", slug: "cool-cat", topic_count: 3234)
+  end
+  fab!(:category2) do
+    Fabricate(:category, name: "Other Category", slug: "other-cat", topic_count: 23)
+  end
+  fab!(:tag) { Fabricate(:tag, name: "cooltag", topic_count: 324) }
+  fab!(:tag2) { Fabricate(:tag, name: "othertag", topic_count: 66) }
+  fab!(:topic) { Fabricate(:topic, category: category, tags: [tag]) }
   fab!(:post) { Fabricate(:post, topic: topic) }
-  fab!(:category) { Fabricate(:category, name: "Cool Category", slug: "cool-cat") }
-  fab!(:tag) { Fabricate(:tag, name: "cooltag") }
+  let(:uncategorized_category) { Category.find(SiteSetting.uncategorized_category_id) }
   let(:topic_page) { PageObjects::Pages::Topic.new }
 
   before do
     SiteSetting.enable_experimental_hashtag_autocomplete = true
     sign_in user
   end
 
-  def visit_topic_and_initiate_autocomplete
+  def visit_topic_and_initiate_autocomplete(initiation_text: "something #co", expected_count: 2)
     topic_page.visit_topic_and_open_composer(topic)
     expect(topic_page).to have_expanded_composer
-    topic_page.type_in_composer("something #co")
+    topic_page.type_in_composer(initiation_text)
     expect(page).to have_css(
       ".hashtag-autocomplete .hashtag-autocomplete__option .hashtag-autocomplete__link",
-      count: 2,
+      count: expected_count,
     )
   end
 
   it "searches for categories and tags with # and prioritises categories in the results" do
     visit_topic_and_initiate_autocomplete
     hashtag_results = page.all(".hashtag-autocomplete__link", count: 2)
-    expect(hashtag_results.map(&:text)).to eq(["Cool Category", "cooltag x 0"])
+    expect(hashtag_results.map(&:text)).to eq(["Cool Category", "cooltag x 325"])
+  end
+
+  it "begins showing results as soon as # is pressed based on categories and tags topic_count" do
+    visit_topic_and_initiate_autocomplete(initiation_text: "#", expected_count: 5)
+    hashtag_results = page.all(".hashtag-autocomplete__link")
```

Comment:
Since we're using this across multiple tests, perhaps we can move this assertion into a page object?

Reply:
Yes will do this when I look at why these system specs are timing out, skipping it for now to fit in with the others